src/irmin-pack/unix/append_only_file.ml

```diff
@@ -52,19 +52,25 @@ module Make (Io : Io.S) = struct
       [open_rw] functions. The [end_offset] from the control file is then used
       as the real offset.
 
-      In case of a crash, if the real offset is larger than the [end_offset], we
-      can safely revert back to [end_offset]. If not, then the store is
-      corrupted, an error is returned. *)
-  let check_consistent_store ~end_offset io =
+      In case of a crash, we can only recover if the [end_offset] is smaller
+      than the real offset. We cannot recover otherwise, because we have no
+      guarantees that the last object fsynced to disk is written entirely to
+      disk. *)
+  let check_consistent_store ~end_offset ~dead_header_size io =
     let open Result_syntax in
     let* real_offset = Io.read_size io in
-    if real_offset < end_offset then Error `Inconsistent_store else Ok ()
+    let dead_header_size = Int63.of_int dead_header_size in
+    let real_offset_without_header =
+      Int63.Syntax.(real_offset - dead_header_size)
+    in
+    if real_offset_without_header < end_offset then Error `Inconsistent_store
+    else Ok ()
```

Comment:
Can you print a warning in the `real_offset_without_header > end_offset` case?

Reply:
done, thanks